### PR TITLE
autoconf adjustments for OpenBSD

### PR DESCRIPTION
**Pull request by sizeofvoid**
- skipping libresolv checks for FreeBSD and OpenBSD.
- improve readline check
  -- In OpenBSD we have two readline libs. On the one hand we have an old
  version in the base OS and on the other hand we have an new one in the
  ports tree. profanity needs the new one and should pick up the right
  one. readline from ports tree is prefixed as ereadline.

### CODE REVIEW

**Comment by jubalh** [comment]
@sizeofvoid can you please rebase these commits so that they only contain your changes without the merges from master?
So we end up with two commits: the pthread include, and the OpenBSD autoconf change.

**Comment by jubalh** [comment]
It seems the patch is still needed for OpenBSD as its still in their tree: http://cvsweb.openbsd.org/cgi-bin/cvsweb/ports/net/profanity/patches/patch-configure_ac

File: configure.ac [hunk 1]
```diff
@@ -172,15 +173,22 @@ AS_IF([test "x$enable_icons" != xno],
...
+              AM_CPPFLAGS="-I/usr/local/opt/readline/include $AM_CPPFLAGS"
+              AM_LDFLAGS="-L/usr/local/opt/readline/lib $AM_LDFLAGS"
+              AC_SUBST(AM_LDFLAGS)])],
+      [test "x$PLATFORM" = xopenbsd],
+      [AC_CHECK_FILE([/usr/local/include/ereadline],
+          [LIBS="-lereadline $LIBS"
+              AM_CPPFLAGS="-I/usr/local/include/ereadline $AM_CPPFLAGS"
+              AM_LDFLAGS="-L/usr/local/lib $AM_LDFLAGS"
+              AC_SUBST(AM_LDFLAGS)])],
+      [test "x$PLATFORM" = xnix],
```
[the remark this answers is not in the document]
Comment (pasis) [reply configure.ac]: You're right, @jubalh. Old logic was:
```
if (osx) { special case }
else { check libreadline }
```
and with the patch it becomes:
```
if (osx) { special case }
else if (openbsd) { special case }
else if (nix) { check libreadline }
else { error }
```
So, the main problem here is in default action (the last `else`). It should be searching for libreadline, which is valid not only for nix. Also, we don't want to rewrite AS_IF macros every time when we add new platform.

@sizeofvoid, fix the default action please. Also I'm not OpenBSD user and can't validate the respective part. Are you sure there is no better way to check existence of `ereadline`? Maybe `pkg-config` or at least `AC_CHECK_LIB`?

**Comment by jubalh** [comment]
@sizeofvoid thanks!

It would be nice to get rid of the merge commits etc and just have two commits.
I tried to cherry pick and rebase.
@sizeofvoid @pasis can you check whether this is now what we want?
https://github.com/boothj5/profanity/compare/master...jubalh:sizeofvoidm?expand=1

If yes I will push to master, close this PR and note down the commits in which it got merged here.

**Comment by pasis** [comment]
@jubalh, sorry for delay, I missed your comment. Personally I would prefer to see a single condition per branch in the AS_IF(). I mean something like this:
```
AS_IF([test "x$PLATFORM" = xosx],
      [osx specific section],
      [x$PLATFORM" = xopenbsd],
      [openbsd specific section],
      [default section with AC_CHECK_LIB]
```
I think this is easier to read and it reduces chance of mistakes. For example, compound condition `test "x$PLATFORM" != xosx -n "x$PLATFORM" != xopenbsd` contains mistake: there must be `-a` instead of `-n`.

Of course you can choose current approach with a little cleanup. I'll leave comments for the rebased patch.

**Comment by jubalh** [comment]
Thanks a lot for checking @pasis.
I updated the commit: https://github.com/boothj5/profanity/compare/master...jubalh:sizeofvoidm

**Comment by pasis** [comment]
@jubalh, there is 1 mistake, check comment for the patch.

**Comment by jubalh** [comment]
Merged with 8fd77ea19bd7441a110b7d0da5fed10ba9be0a2c and bcaf55e5b8525ea91346657520ff47f45cbbe064.

@sizeofvoid thanks for patch!
@pasis thanks very much for detailed instructions on how to change it!

**Pull request closed without merging**